pyxis/pyxis.go

```diff
@@ -264,14 +327,20 @@ func syncRepo(repo models.Repository) error {
 				return err
 			}
 		}
+		tags, _ := extractTags(apiRepoImages[pyxisID])
 		if _, found := dbRepositoryImageMap[image.ID]; !found {
 			// repository_image pair not found
 			toInsertRepositoryImages = append(
 				toInsertRepositoryImages,
-				models.RepositoryImage{
-					RepositoryID: repo.ID,
-					ImageID:      image.ID,
-				},
+				newRepoImage(repo.ID, image.ID, tags),
+			)
+		} else if repImg, _ := dbRepositoryImageMap[image.ID]; checkTags(repImg, tags) {
```

Comment:
This branch should also contain:
```
delete(dbRepositoryImageMap, image.ID)
```

same as the `else` branch, otherwise the updated repository-image rows will be marked for deletion on lines 351-353.